handsontable/src/plugins/dialog/dialogUI.js

```diff
@@ -0,0 +1,229 @@
+   *
+   * @param {object} options - Class name update options.
+   * @param {boolean} options.isVisible - Whether the dialog is visible.
+   * @param {string|HTMLElement} options.content - The content to render in the dialog.
+   * @param {string} options.customClassName - The custom class name to add to the dialog.
+   * @param {string} options.background - The background to add to the dialog.
+   * @param {boolean} options.contentBackground - Whether to show content background.
+   * @param {string} options.contentDirections - The flex direction for content layout.
+   * @param {boolean} options.animation - Whether to add the animation class to the dialog.
+   *
+   * @returns {DialogUI} The instance of the DialogUI for method chaining.
+   */
+  updateDialog({
+    isVisible,
+    content,
+    customClassName,
+    background,
+    contentBackground,
+    contentDirections,
+    animation,
+  }) {
+    // Dialog class name
+    const customClass = customClassName ?
+      ` ${customClassName}` : '';
+    const backgroundClass = background ?
+      ` ${DIALOG_CLASS_NAME}--background-${background}` : '';
+    const animationClass = animation ?
+      ` ${DIALOG_CLASS_NAME}--animation` : '';
+    const showClass = isVisible ? ` ${DIALOG_CLASS_NAME}--show` : '';
+
+    // Update dialog class name
+    this.#refs.dialogElement.className =
+      `${DIALOG_CLASS_NAME}${customClass}${backgroundClass}${animationClass}${showClass}`;
+
+    // Dialog content class name
+    const contentBackgroundClass = contentBackground ?
+      ` ${DIALOG_CLASS_NAME}__content--background` : '';
+    const contentDirectionsClass = contentDirections ?
+      ` ${DIALOG_CLASS_NAME}__content--flex-${contentDirections}` : '';
+
+    // Update content class name
+    this.#refs.contentElement.className =
+      `${DIALOG_CLASS_NAME}__content${contentBackgroundClass}${contentDirectionsClass}`;
+
+    // Clear existing dialog content
+    this.#refs.contentElement.innerHTML = '';
+
+    // Render new dialog content
+    if (typeof content === 'string') {
+      fastInnerHTML(this.#refs.contentElement, content);
+    } else if (content instanceof HTMLElement || content instanceof DocumentFragment) {
+      this.#refs.contentElement.appendChild(content);
+    }
+
+    return this;
+  }
+
+  /**
+   * Shows the dialog with optional animation.
+   *
+   * @param {boolean} animation - Whether to add the animation class to the dialog.
+   * @returns {DialogUI} The instance of the DialogUI for method chaining.
+   */
+  showDialog(animation) {
+    this.#refs.dialogElement.style.display = 'block';
+
+    if (animation) {
+      // Triggers style and layout recalculation, so the display: block is fully committed before adding
+      // the class ht-dialog--show.
+      // eslint-disable-next-line no-unused-expressions
+      this.#refs.dialogElement.offsetHeight;
+    }
+
+    addClass(this.#refs.dialogElement, `${DIALOG_CLASS_NAME}--show`);
+
+    return this;
+  }
```

Comment:
I propose assigning the ref to a variable. Thanks to that, the code will be more optimal for space compression.
```suggestion
  showDialog(animation) {
    const { dialogElement } = this.#refs;

    dialogElement.style.display = 'block';

    if (animation) {
      // Triggers style and layout recalculation, so the display: block is fully committed before adding
      // the class ht-dialog--show.
      // eslint-disable-next-line no-unused-expressions
      dialogElement.offsetHeight;
    }

    addClass(dialogElement, `${DIALOG_CLASS_NAME}--show`);

    return this;
  }
```